cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/CassandraScheduler.java

```diff
@@ -141,6 +142,12 @@ public CassandraScheduler(
     public void registered(SchedulerDriver driver,
                            Protos.FrameworkID frameworkId,
                            Protos.MasterInfo masterInfo) {
+        // Return early if registered() has already been called once in the lifetime of this object.
+        if (!isSchedulerRegistered.compareAndSet(false, true)) {
```

Comment:
I am inclined to just call reregistered() in this case. Does that sound ok @mohitsoni ?